ipaserver/install/server/install.py

```diff
@@ -474,6 +474,11 @@ def install_check(installer):
     else:
         realm_name = options.realm_name.upper()
 
+    try:
+        validate_domain_name(realm_name)
+    except ValueError as e:
+        raise ScriptError("Invalid realm name: {}".format(unicode(e)))
+
```

Comment:
This check is performed quite late in the installation. You may consider adding
```
    @realm_name.validator
    def realm_name(self, value):
        validate_domain_name(value)
```
into ipalib/install/service.py so that the validation is performed earlier (as for domain_name). Keep the validation here for prompted input, though.

It would also be more clear if the error message for realm mentioned 'single label realms are not supported' instead of 'single label domains'.

Reply:
Thank you so much for pointing this out. Didn't know about validators in `ipalib/install/service.py` :)